packages/server/test/regression/issue-949-inferProcedureInput.test.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+// IMPORTANT:
+// needs to be imported from compiled output otherwise we get a false-positive
+import * as trpc from '../../dist/trpc-server.cjs';
+import { expectTypeOf } from 'expect-type';
```

Comment:
Interstingly in debugging this I found that there's _also_ a bug in expect-type:

```
import { expectTypeOf } from 'expect-type'

class Foo {
  constructor(input: string) {}
}

class Bar {
  constructor(input: number) {}
}

expectTypeOf<Foo>().not.toEqualTypeOf<Bar>() // fails!
```

...so I'll need to push an update to that too.